src/sidebar/components/user-menu.js

```diff
@@ -11,15 +11,32 @@ import MenuItem from './menu-item';
 import MenuSection from './menu-section';
 import SvgIcon from '../../shared/components/svg-icon';
 
+/** @typedef {import('../../types/config').SidebarConfig} SidebarConfig */
+
+/** @typedef {import('../../types/config').HostConfig} HostConfig */
+
+/** @typedef {import('../../types/auth').AuthState} AuthState */
+
+/**
+ * @typedef UserMenuProps
+ * @prop {AuthState} auth - object representing authenticated user and auth status
+ * @prop {() => any} onLogout - onClick callback for the "log out" button
+ * @prop {Object} bridge
+ * @prop {(name: string, params?: Object<string, string>) => any} serviceUrl
```

Comment:
There is a `ServiceUrlGetter` type in `src/sidebar/services/service-url.js` that you can use here for the `serviceUrl` prop.